File: Common-Modules/src/main/java/com/common/dto/UserResponseDTO.java
package com.common.dto;

import java.util.List;
import java.util.Map;

import com.common.entity.SaloonService;

import lombok.Data;

@Data
public class UserResponseDTO {

	private String keycloakUserId;
	private String firstName;
	private String lastName;
	private String email;
	private boolean enable;
	private Map<String, List<String>> attributes;
	private List<SaloonService> bookedServiceIds;
}

File: src/app/services/DTOs/user-response-dto.ts
import { SaloonServiceDTO } from "./saloon-service-dto";

export interface UserResponseDTO {
  keycloakUserId: string;
  firstName: string;
  lastName: string;
  email: string;
  enable: boolean;
  attributes?: {
    [key: string]: string[];
  };
  bookedServiceIds: SaloonServiceDTO[];
}